src/main/groovy/io/seqera/wave/service/logs/BuildLogServiceImpl.groovy

```diff
@@ -152,8 +152,27 @@ class BuildLogServiceImpl implements BuildLogService {
         if( !logs ) return
         try {
             String condaLock = extractCondaLockFile(logs)
-            if (condaLock){
+            if ( condaLock ){
                 log.debug "Storing conda lock for buildId: $buildId"
+
+                /* When a container image is cached, dockerfile does not get executed.
+                 In that case condalock file will contain "cat environment.lock" because its not been executed.
+                 So wave will check the previous builds of that container image
+                 and render the condalock file from latest successful build
+                 and replace with the current build's condalock file.
+                 */
+                if( condaLock.contains('cat environment.lock') ){
+                    log.info "Container Image is already cached, uploading previously successful build's condalock file for buildId: $buildId"
+                    def builds = persistenceService.allBuilds(buildId.split('-')[1].split('_')[0])
+                    for (def build : builds) {
+                        if ( build.succeeded() ){
+                            def curCondaLock = fetchCondaLockString(build.buildId)
+                            if( curCondaLock && !curCondaLock.contains('cat environment.lock') ){
+                                condaLock = curCondaLock
+                            }
+                        }
+                    }
+                }
```

Comment:
let's isolate this logic into a separate method, so it can be tested independently

Reply:
done